.github/workflows/release_python_nightly.yml

```diff
@@ -0,0 +1,113 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: "Nightly PyPI Build"
+
+on:
+  schedule:
+    - cron: "0 0 * * *"  # Runs at midnight UTC every day
+  workflow_dispatch:  # Allows manual triggering
+
+env:
+  rust_msrv: "1.77.1"
+
+permissions:
+  contents: read
+
+jobs:
+  sdist:
+    if: github.repository == 'apache/iceberg-rust'  # Only run for apache repo
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ./.github/actions/overwrite-package-version  # overwrite the pacakge version with current timestamp
+
+      - uses: PyO3/maturin-action@v1
+        with:
+          working-directory: "bindings/python"
+          command: sdist
+          args: -o dist
+      - name: Upload sdist
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-sdist
+          path: bindings/python/dist
+
+  wheels:
+    if: github.repository == 'apache/iceberg-rust'  # Only run for apache repo
+    runs-on: "${{ matrix.os }}"
+    strategy:
+      matrix:
+        include:
+          - { os: windows-latest }
+          - { os: macos-latest, target: "universal2-apple-darwin" }
+          - { os: ubuntu-latest, target: "x86_64" }
+          - { os: ubuntu-latest, target: "aarch64" }
+          - { os: ubuntu-latest, target: "armv7l" }
+    steps:
+        - uses: actions/checkout@v4
+        - uses: ./.github/actions/overwrite-package-version  # overwrite the pacakge version with current timestamp
+        - uses: actions/setup-python@v5
+          with:
+            python-version: 3.9
+        - name: Setup Rust toolchain
+          uses: ./.github/actions/setup-builder
+          with:
+            rust-version: ${{ env.rust_msrv }}
+        - uses: PyO3/maturin-action@v1
+          with:
+            target: ${{ matrix.target }}
+            manylinux: auto
+            working-directory: "bindings/python"
+            command: build
+            args: --release -o dist
+          env:
+            # Workaround ring 0.17 build issue
+            CFLAGS_aarch64_unknown_linux_gnu: "-D__ARM_ARCH=8"
```

Comment:
This workaround is wrong and need to fix. Please refer to https://github.com/apache/opendal/pull/5522 for changes, and https://github.com/apache/opendal/issues/5483 for details.

Reply:
I copied this over from the original `release_python.yml`
https://github.com/apache/iceberg-rust/blob/562ec947df35d172306c078fc42777f6d10b060f/.github/workflows/release_python.yml#L84-L85

Let me open a fix separately and then rebase on that change

Reply:
#975

Reply:
also applied the same change to this PR